data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraServiceTest.java

```diff
@@ -0,0 +1,315 @@
+        SearchResults mockSearchResults = mock(SearchResults.class);
+        doReturn(new ResponseEntity<>(mockSearchResults, HttpStatus.OK)).when(restTemplate).getForEntity(any(URI.class), any(Class.class));
+        SearchResults results = jiraService.getAllIssues(jql, 0, jiraSourceConfig);
+        assertNotNull(results);
+    }
+
+    private JiraSourceConfig createJiraConfiguration(String auth_type, List<String> issueType, List<String> issueStatus, List<String> projectKey) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, String> connectorCredentialsMap = new HashMap<>();
+        connectorCredentialsMap.put("auth_type", auth_type);
+
+        Map<String, Object> jiraSourceConfigMap = new HashMap<>();
+        jiraSourceConfigMap.put("account_url", ACCESSIBLE_RESOURCES);
+        jiraSourceConfigMap.put("connector_credentials", connectorCredentialsMap);
+        jiraSourceConfigMap.put("issue_type", issueType);
+        jiraSourceConfigMap.put("status", issueStatus);
+        jiraSourceConfigMap.put("project", projectKey);
+
+
+        String jiraSourceConfigJsonString = objectMapper.writeValueAsString(jiraSourceConfigMap);
+        return objectMapper.readValue(jiraSourceConfigJsonString, JiraSourceConfig.class);
+    }
+
+    private IssueBean createIssueBean(boolean nullFields) {
+        IssueBean issue1 = new IssueBean();
+        issue1.setId(UUID.randomUUID().toString());
+        issue1.setKey("issue_1_key");
+        issue1.setSelf("https://example.com/rest/api/2/issue/123");
+        issue1.setExpand("operations,versionedRepresentations,editmeta");
+
+        Map<String, Object> fieldMap = new HashMap<>();
+        if (!nullFields) {
+            fieldMap.put(CREATED, Instant.now());
+            fieldMap.put(UPDATED, Instant.now());
+        } else {
+            fieldMap.put(CREATED, 0);
+            fieldMap.put(UPDATED, 0);
+        }
+
+        Map<String, Object> issueTypeMap = new HashMap<>();
+        issueTypeMap.put("name", "Task");
+        issueTypeMap.put("self", "https://example.com/rest/api/2/issuetype/1");
+        issueTypeMap.put("id", "1");
+        fieldMap.put("issuetype", issueTypeMap);
+
+        Map<String, Object> projectMap = new HashMap<>();
+        if (!nullFields) {
+            projectMap.put("name", "project name test");
+            projectMap.put(KEY, "TEST");
+        }
+        fieldMap.put("project", projectMap);
+
+        Map<String, Object> priorityMap = new HashMap<>();
+        priorityMap.put("name", "Medium");
+        fieldMap.put("priority", priorityMap);
+
+        Map<String, Object> statusMap = new HashMap<>();
+        statusMap.put("name", "In Progress");
+        fieldMap.put("status", statusMap);
+
+        issue1.setFields(fieldMap);
+
+        return issue1;
+    }
+
+    private void waitForFutures(List<Future<Boolean>> futureList) throws InterruptedException {
```

Comment:
We must have a time limit here. Otherwise this could be waiting for ever

Reply:
We removed this method now